pkg/store/proxy.go

```diff
@@ -342,6 +351,12 @@ func (s *ProxyStore) Series(originalRequest *storepb.SeriesRequest, srv storepb.
 		}
 	}
 
+	if respHeap.UnsortedSetDetected() {
```

Comment:
This feels a bit like implementing InfoServer through the Series service. What do you think about checking `		st.SendsSortedSeries()` instead and if all are true then we could also advertise that `SendsSortedSeries() == true` in Thanos Query. And if there is at least one where `SendsSortedSeries() == false` then let's also advertise `false` in Thanos Query. 

I have been playing around with your PR a little bit and :point_up: seems like a much cleaner solution. I can try opening up a PR to it tomorrow or on Wednesday.

Reply:
Now that I read that I again, I am not sure if it is possible with the current implementation of stores to make a detection statically without inspecting the data. A user can specify any label as the replica label in a Query component. So stores need to resort data arbitrarily in the Series call if we want to have checks through info calls.